source/docs/software/commandbased/commands-v3/mechanisms.rst

```diff
@@ -0,0 +1,70 @@
+# Mechanisms
+
+.. todo::
+   This article provides a comprehensive guide to Mechanisms, the v3 replacement for Subsystems. It explains the philosophy behind Mechanisms, how to create them, best practices for encapsulation, and how they interact with the command system. The focus should be on teaching teams how to properly structure robot hardware abstractions in v3.
+
+## Creating Mechanisms
+
+.. todo::
+   Describe how to create a Mechanism class. Cover:
+   - Basic Mechanism class structure and constructor
+   - How Mechanisms differ philosophically from v2 Subsystems
+   - When to create a Mechanism vs. other organizational patterns
+   - Naming conventions for Mechanism classes
+   - Where Mechanism classes should live in the project structure
+   - How to register Mechanisms with the command scheduler (if needed)
+
+## Encapsulation Best Practices
+
+.. todo::
+   Explain encapsulation principles for Mechanisms. Cover:
+   - What hardware and state should be encapsulated within a Mechanism
+   - Public API design: what methods should be exposed vs. kept private
+   - How to avoid leaking implementation details
+   - Sensor data processing and filtering within Mechanisms
+   - State management and validation
+   - Examples of good vs. bad encapsulation patterns
+
+## Default Commands
+
+.. todo::
+   Explain how default commands work with Mechanisms. Cover:
+   - What default commands are and when to use them
+   - How to set a default command for a Mechanism
+   - Lifecycle of default commands (when they start/stop)
+   - Common use cases (idle states, teleoperated control, etc.)
+   - How default commands interact with other commands requiring the same Mechanism
+   - Best practices for implementing default command behavior
+
+## Periodic Updates
+
+.. todo::
+   Describe periodic update patterns in Mechanisms. Cover:
+   - Whether Mechanisms have a periodic() method like v2 Subsystems
```

Comment:
Well, they don't. But manual periodic functions can plumbed if needed, or just called manually in `robotPeriodic()`

```java
class Drive extends Mechanism {
  void refreshSensors() {
    // ...
  }
}

class Robot extends TimedRobot {
  Drive drive;
  
  // With Scheduler.addPeriodic:
  public Robot() {
    Scheduler.getDefault().addPeriodic(drive::refreshSensors);
  }
  
  // Or just call the function manually:
  @Override
  public void robotPeriodic() {
    drive.refreshSensors();
    Scheduler.getDefault().run();
  }
}
```